presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.esri.core.geometry.Envelope;
+import com.esri.core.geometry.MultiPath;
+import com.esri.core.geometry.MultiPoint;
+import com.esri.core.geometry.OperatorContains;
+import com.esri.core.geometry.OperatorCrosses;
+import com.esri.core.geometry.OperatorDisjoint;
+import com.esri.core.geometry.OperatorEquals;
+import com.esri.core.geometry.OperatorIntersects;
+import com.esri.core.geometry.OperatorOverlaps;
+import com.esri.core.geometry.OperatorSimpleRelation;
+import com.esri.core.geometry.OperatorTouches;
+import com.esri.core.geometry.OperatorWithin;
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.Polygon;
+import com.esri.core.geometry.SpatialReference;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.esri.core.geometry.ogc.OGCLineString;
+import com.esri.core.geometry.ogc.OGCMultiLineString;
+import com.esri.core.geometry.ogc.OGCPoint;
+import com.esri.core.geometry.ogc.OGCPolygon;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import org.apache.commons.lang.StringUtils;
+
+import java.lang.reflect.Method;
+
+import static com.facebook.presto.geospatial.GeometryUtils.deserialize;
+import static com.facebook.presto.geospatial.GeometryUtils.serialize;
+import static java.util.Arrays.asList;
+
+public final class GeoFunctions
+{
+    private static final String LINE = "LineString";
```

Comment:
GeometryUtils.LINE_STRING to avoid copy-pasting hard-coded strings

Ditto other constants.